### PR TITLE
added dodge function and did some refactoring

### PR DESCRIPTION
**Pull request by bergerm613**

### CODE REVIEW

File: src/main/java/coloringpage/Converter.java [hunk 1]
```diff
@@ -2,69 +2,113 @@
...
-        float[] matrix = new float[400];
-        for (int i = 0; i < 400; i++) {
-            matrix[i] = 1.0f / 400.0f;
-        }
-
-        int blurRad = 300;
-        float[] blurKernel = new float[blurRad*blurRad];
-        for(int i =0; i<blurRad*blurRad; i++) {
-            blurKernel[i] = 1.0f/256.0f;
-        }
```
Comment (bergerm613) [inline src/main/java/coloringpage/Converter.java:46]: @Esther sheina, it looked like this code didn't do anything, so I deleted it, confirm?

Comment (ester-agishtein) [reply src/main/java/coloringpage/Converter.java:46]: NOOO!!!!
That's the blur function!!! That's how you set the blurriness!

File: src/main/java/coloringpage/Converter.java [hunk 2]
```diff
@@ -2,69 +2,113 @@
...
+
         int radius = 20;
         int size = radius * 2 + 1;
         float weight = 1.0f / (size * size);
         float[] data = new float[size * size];
 
-        for (int i = 0; i < data.length; i++) {
-            data[i] = weight;
-        }
+        Arrays.fill(data, weight);
```
Comment (ester-agishtein) [inline src/main/java/coloringpage/Converter.java:77]: This is fine as a replacement for my for loop. I assume it basically does the same thing.

File: src/main/java/coloringpage/Converter.java [hunk 3]
```diff
@@ -2,69 +2,113 @@
...
+        int width = front.getWidth();
+        int height = front.getHeight();
+
+        for(int i=0; i<height; i++) {
+            for(int j=0; j<width; j++) {
+                int frontValue = new Color(front.getRGB(j, i)).getRed(); //could be any of RGB, red is arbitrary
+                int backValue = new Color(back.getRGB(j, i)).getRed();
+
+                int newValue = frontValue * 255 / (255 - backValue);
+                if (newValue > 255 || backValue == 255) {
```
Comment (ester-agishtein) [inline src/main/java/coloringpage/Converter.java:95]: Oh is this how you're checking for the blue/inverting black issue?

File: src/main/java/coloringpage/Converter.java [hunk 4]
```diff
@@ -2,69 +2,113 @@
...
+            }
+        }
+
+        return result;
+    }
+
+    private BufferedImage cloneImage(BufferedImage image) {
+        //https://bytenota.com/java-cloning-a-bufferedimage-object/
+        ColorModel colorModel = image.getColorModel();
+        WritableRaster raster = image.copyData(null);
```
Comment (ester-agishtein) [inline src/main/java/coloringpage/Converter.java:110]: What is a WritableRaster?

File: src/main/java/coloringpage/Converter.java (hunk 1 above)
Comment (bergerm613) [reply src/main/java/coloringpage/Converter.java:46]: so thats what I thought, but then the function `blurImage()` (which I renamed from `processImage()`) does that as well, and this chuck of code isn't used

File: src/main/java/coloringpage/Converter.java (hunk 2 above)
Comment (bergerm613) [reply src/main/java/coloringpage/Converter.java:77]: aye

File: src/main/java/coloringpage/Converter.java (hunk 3 above)
Comment (bergerm613) [reply src/main/java/coloringpage/Converter.java:95]: nopes, this is just the code to merge layers of the pic together, to make the final line drawing. but I should add a javadoc comment explaining that

File: src/main/java/coloringpage/Converter.java (hunk 4 above)
Comment (bergerm613) [reply src/main/java/coloringpage/Converter.java:110]: apparently there is no easy way to make a deep copy of a BufferedImage, so this is a method I found to copy each part of it individually and put them together. I believe the WriteableRaster part has to do with the actual pixels in the image

**Pull request merged**